### ISSUE TITLE
the battery is drained twice as fast

### ISSUE THREAD

**Issue by Free-Inside** [ubports/ubuntu-touch#1809]
- Device:OnePlus One
- Channel:RC
- Build:2021-W33

### Steps to reproduce

I don't know what produces it

### Expected behavior

the battery is draining normally (about 2-3 days)

### Actual behavior

the battery is drained in less than 24 hours with moderate use

### Logfiles and additional information

even with the latest RC the battery still drains fast, even with several reboots. To be sure I reinstalled UT in RC with the UBports installer without the wipe option, and it didn't change anything

**Comment by mfoucrier** [comment]
Since the last update, OTA-18, the battery of the Meizu Pro5 is draining almost 2 times faster! I have the impression that the BQ 4.5 has not been affected? Observation made since a few weeks in usual use.

**Comment by lduboeuf** [comment]
same situation here : https://forums.ubports.com/topic/6589/batter-life-since-last-ota/7

**Comment by Flohack74** [comment]
BQ E4.5 is also affected according to a user in my family ;)

**Comment by Free-Inside** [comment]
> same situation here : https://forums.ubports.com/topic/6589/batter-life-since-last-ota/7

I don't think on my OnePlus One it's the same situation, but indeed on my BQ M10FHD the battery goes down in spurts, and a Kworker command comes back regularly and uses a lot of CPU like on the link pictures, I thought it was my battery

**Comment by Free-Inside** [comment]
On my OnePlus One there is Binder and Sensors.qc which are used all the time, I have checked several times, I even monitored it for 10 minutes and it never stops.
Is this normal? 
And could it be that this is what is preventing the deep sleep and therefore the battery discharge?
![screenshot](https://user-images.githubusercontent.com/89270281/131229904-4f7c3d56-aaae-4280-9b12-47eac997812d.png)

**Comment by p3nguyn** [comment]
I have a similar problem with my E4.5. If I deactivate Wifi and mobile data and do a reboot after that the battery life is normal (ca. 2-3 days for my usage). Once I activate network connections it may take a while but at some point the battery drains very fast. I can even feel that my phone is warm. The battery emptys within ca. 10-15 hours.
What happens if you deactivate network data and do a reboot? Is the battery life normal again?

**Comment by Free-Inside** [comment]
I did this without restarting the phone and yes the battery consumption is back to normal. I had done some testing to try and identify where the problem might be coming from.

Which does not change the over consumption :
- with cellular data + without wifi and turn off all notifications from the settings
- with cellular data + with wifi and with notification
- without cellular data + with wifi and with notification
- without cellular data + with wifi and without notification

Normal consumption :
- without cellular data + without wifi
- with 2G cellular data with notification and without wifi

Overconsumption cannot come from 3G or 4G research because even without cellular data with wifi and without notification the battery discharges more quickly. But there is something on the network in wifi and 3/4G that consumes the battery and/or prevents deep sleep

**Comment by Flohack74** [comment]
Please try to find out the process that consumes all CPU in this moment, when the drain is high and when the phone feels warm: Open terminal, rotate to landscape and then type top and lets see what you can find :)

**Comment by p3nguyn** [comment]
> 
> 
> Please try to find out the process that consumes all CPU in this moment, when the drain is high and when the phone feels warm: Open terminal, rotate to landscape and then type top and lets see what you can find :)

At the moment my phone does not feel hot but the battery drains way to wast anyways. Here is the requestet screenshot:
![grafik](https://user-images.githubusercontent.com/90145807/132141332-26247ac4-edf6-4051-90f1-bbd6533ee92b.png)

And here you can see the battery-graph:
![grafik](https://user-images.githubusercontent.com/90145807/132141397-d0bd4815-5112-49fb-851f-f91977508684.png)
At about 17:00 I turned on the network connection.

**Comment by Flohack74** [comment]
Humm that does not give much results. Maybe modem does not go to hibernate anymore? I have no clue if its not a process actually

**Comment by p3nguyn** [comment]
Now my phone feelt hot.
![grafik](https://user-images.githubusercontent.com/90145807/132558480-1ec7731d-465b-49f9-81a1-657598336592.png)

This is the overview of processes right after running the top command at the time with the high power usage:
![grafik](https://user-images.githubusercontent.com/90145807/132556964-3668f479-3e14-4d96-be45-c777464aa856.png)

A few seconds after running the top command the CPU usage drops down to something like this:
![grafik](https://user-images.githubusercontent.com/90145807/132557232-98494044-76a8-46e5-9c63-d28030b2e5d8.png)

I tried sending a picture via telegram with a bad mobile network signal. Telegram was not able to send the pictures. As soon as I activated Wifi telegram was able to send the pictures. Now my phone cooled down again. As you can see in the first picture the rapid discharging stopped right after sucessfully sending the picture.

Can someone confirm a connection between telegram unsuccesfully trying to send a picture and the massive battery drainage?

**Comment by Flohack74** [comment]
Unfortunately we do not see the process names here. The display is simply too small. Teleports can utilize quite some CPU, but it is suspended in the background, so its not the thing that will drain your battery when you dont use it.

**Comment by Free-Inside** [comment]
When I do a "top" in the terminal and I lock the screen with the power button or I let the screen turn off automatically, when I wake up the phone 5 to 10 minutes later, the "top" command in the terminal continues to run instead of stopping automatically when the screen locks.

**Comment by Flohack74** [comment]
@Free-Inside you mean connected via ssh or adb?

**Comment by Free-Inside** [comment]
> @Free-Inside you mean connected via ssh or adb?

no it is not connected to the PC, I tried several times and it does not do it every time

**Comment by girlwithnoname** [comment]
Well, I have some battery drain, too. Firing top via SSH shows cnss_diag as top most process (2-5% CPU) on a Oneplus 6T on devel channel. Second process is biometryd. The latter is spamming my system logs. Can I disable this process somehow as fingerprint isn't yet working anyway?

**Comment by SolAZDev** [comment]
I happen to have a similar issue, I'm working on a port (so I apologize if Im doing this wrong) but the others with the same SoC seem to have the issue; where `unity-system-compositor` takes 100% of CPU.

Device: Samsung Galaxy S9+ (Other devices are Galaxy S9 and Note 9)

![image](https://user-images.githubusercontent.com/6070602/133019211-d5286195-738f-4a7d-a37c-f70bc23fda19.png)

**Comment by Flohack74** [comment]
Well u-s-c should only take CPU when the screen is active, is this the case during that CPU load?

**Comment by SolAZDev** [comment]
@Flohack74 in my case, it is in fact using cpu when the screen is on, and yes, its CPU Usage goes to 100%

**Comment by Flohack74** [comment]
@SolAZDev and when you turn it off, it returns to normal? Still, it should not do 100% continuously, amybe spike a bit when needed.

**Comment by girlwithnoname** [comment]
I have to correct myself: the drain is still there. The processes cnss_diag and biometryd are slowly eating my battery (at a rate of 2% per hour if I don't touch the device). Is this device specific? Cnss_diag seems to be disable in /android/vendor/etc/init/hw/init.qcom.rc, nevertheless it's still running. Can I somehow work around the drain?

**Comment by Flohack74** [comment]
Yes thats purely device-specific. cnss_diag is a nasty thing, but I had something something seen to disable it. However, are you sure biometryd is also eating battery? It should be entirley dormant until you use your FP reader

**Comment by SolAZDev** [comment]
@Flohack74 Unfortunately, that is not the case. 
This is top running with the screen off
![image](https://user-images.githubusercontent.com/6070602/133366909-7ed2cf5a-1518-41b4-92ea-b4e13cbc6c57.png)

**Comment by girlwithnoname** [comment]
@Flohack74 Well, biometryd is crashing and respawning the whole time. I saw it in dmesg, other logs are disabled it seems. I helped myself by remounting / rw and renaming cnss_diag and biometryd and now they aren't running anymore. I made a bunch of modifications yesterday evening and what helped the most was to disable network while the screen is off. Now it is <1% drain per hour.

**Comment by Flohack74** [comment]
crashing biometryd: Do you have a fingerprint reader on your device?

**Comment by girlwithnoname** [comment]
Yes. Unfortunately it's an under-display fingerprint reader so I don't think it'll work soon.

**Comment by Flohack74** [comment]
Can you obtain a logcat log from the Android container? We might see why the underlying API is crashing

**Comment by girlwithnoname** [comment]
How would I do that?Adb doesn't work on my device.

**Comment by girlwithnoname** [comment]
So I fired adb logcat and this is what I got:
~$ adb logcat
/bin/bash: line 0: exec: logcat: not found

**Comment by girlwithnoname** [comment]
dmesg shows a ton of these:
[    0.000000] init: biometryd main process ended, respawning
[    0.000000] init: biometryd main process (9394) killed by ABRT signal
[    0.000000] init: biometryd main process ended, respawning
[    0.000000] init: biometryd main process (9413) killed by ABRT signal
[    0.000000] init: biometryd main process ended, respawning
[    0.000000] init: biometryd main process (9435) killed by ABRT signal
[    0.000000] init: biometryd main process ended, respawning
[    0.000000] init: biometryd main process (9470) killed by ABRT signal
[    0.000000] init: biometryd main process ended, respawning
[    0.000000] init: biometryd main process (9525) killed by ABRT signal
[    0.000000] init: biometryd main process ended, respawning
[    0.000000] init: biometryd main process (9634) killed by ABRT signal
[    0.000000] init: biometryd main process ended, respawning
[    0.000000] init: biometryd main process (9686) killed by ABRT signal
[    0.000000] init: biometryd main process ended, respawning

**Comment by mfoucrier** [comment]
Hello,
I hope this can be useful...
Since OTA18, when I'm on the move (on a train for example) the cell phone heats up much more when using 3G/4G than before, there has been a change in the 3G/4G stack. Yes, in airplane mode during the night the battery discharges a little faster than before too, but it is much more important during the day when using. Thank you for your work.

**Comment by Free-Inside** [comment]
Does this remove IPv6 requests for the mobile network ?
https://pad.ubports.com/p/nm-battery-experiment/timeslider#692

I've tried it on my OnePlus One for 2 weeks and I have no difference on battery consumption, but in the comments there is a person who says that it still receives an IPv6 even after the change. 
Does it really cut the IPv6 ?
Is there anything we can do to turn off IPv6 and see if the problem is caused by that ?

**Comment by donbubu** [comment]
Hi,
same problem on Fairphone 3+. Battery down within 27h, all off and cpu on 20%.

Edit: With connection "2g" and "2g/3g" it work over 90h , but with "2g/3g/4g" 27h and empty
